Userland/Libraries/LibJS/Runtime/Value.cpp

```diff
@@ -81,7 +81,7 @@ static String double_to_string(double d)
         builder.append(double_to_string(-d));
         return builder.to_string();
     }
-    if (d == INFINITY)
+    if (d == static_cast<double>(INFINITY))
```

Comment:
as `INFINITY` is supposed to be a float, please update our `math.h` as well where it's currently defined as `__builtin_huge_val()`, which is a double according to the GCC docs.